chapter8.adoc

```diff
@@ -0,0 +1,37 @@
+[[chapter8]]
+[[Smsdedbg]]
+== `Smsdedbg`: Supervisor Domain External Debug
+
+[caption="Figure {counter:image}: ", reftext="Figure {image}"]
+[title= "External Debug for Supervisor Domain", id=Smsdedbg_img]
+image::Smsdedbg.png[]
+
+The <<MSDCFG>> CSR contains the `sdedbgalw` bit that controls whether the
+current scheduled SD is allowed to be external-debugged. This bit is context
+switched (along with rest of the `msdcfg`) per SD.
+
+=== `Smsdedbg` interaction with external debug security controls (Informative)
+
+This section will be moved into the non-ISA specification for external debug
+security. It is described here as informational.
+
+The `medbgen` is an enable control for external debug for the M-mode driven by
+the debug module and is expected to be established by the RoT (following RISC-V
+Security Model recommendation SR_GEN_007 and SR_GEN_012). When privilege is `M`,
+the `medbgen` gates the `haltreq_i` from the debug module and if is 0 prevents
+the hart from entering external debug mode. When privilege is less than `M`, the
+OR of the `MSDCFG.sdedbgalw` and `medbgen` gates the `haltreq_i` from the debug
+module and the hart will enter external debug mode if either is 1.
+
+The configuration for `MSDCFG.sdedbgalw` may be obtained from the manifest/
+configuration of the supervisoer domain and should be managed by the M-mode root
+security manager using secure memory.
+
+When `medbgen` is 1, there are no restrictions. When `medbgen` is 0 but
+`MSDCFG.sdedbgalw` is 1, then the external debug mode may be entered but has
+to have restrictions such as `M-mode` -privileged CSR accesses are prohibited
+and Memory accesses by instructions executed in external debug mode cannot
+use `M` privilege. When `medbgen` is 0 and `MSDCFG.sdedbgalw` is also 0, then
+the rules around triggers configured with action = 1 that may fire - such as
+always cause breakpoint exception. The external debug security extension TG
+will need to define the rules around what is allowed in external debug mode.
```

Comment:
```suggestion
When `medbgen` is 1, there are no restrictions. When `medbgen` is 0 but
`MSDCFG.sdedbgalw` is 1, then the external debug mode may be entered but is
restricted by `M-mode` software to prevent privileged CSR accesses and memory
accesses by instructions executed in external debug mode cannot use `M` privilege.
When `medbgen` is 0 and `MSDCFG.sdedbgalw` is also 0, when sdebgalw is 0, the
M-mode root domain security manager must not configure triggers with `action=1`.
```